### PR TITLE
audit whether prerelease or not

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?
-----
Closes https://github.com/Homebrew/brew/issues/7177
Check projects on GitHub whether pre-release or not.

### CODE REVIEW

File: Library/Homebrew/dev-cmd/audit.rb [hunk 1]
```diff
@@ -795,6 +795,14 @@ def audit_specs
         return if stable_url_minor_version.even?
 
         problem "#{stable.version} is a development release"
+      when %r{^https://github.com/}
+        repo_url = URI.parse(stable.url)
+        path = repo_url.path.split("/")
+        owner = path[1]
+        repo = path[2]
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/audit.rb]:
```suggestion
        _, owner, repo, = URI.parse(stable.url).path.split("/")
```

File: Library/Homebrew/dev-cmd/audit.rb [hunk 2]
```diff
@@ -795,6 +795,14 @@ def audit_specs
...
+        if GitHub.open_api("#{GitHub::API_URL}/repos/#{owner}/#{repo}/releases/tags/#{stable.version}")["prerelease"]
+          problem "#{stable.version} is a pre release"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/audit.rb]:
```suggestion
          problem "#{stable.version} is a GitHub prerelease"
```

**Comment by MikeMcQuaid** [comment]
Looks good! Do you have any prereleases you've found that I can use to test this?

**Comment by hyuraku** [comment]
> Looks good! Do you have any prereleases you've found that I can use to test this?

I audited [sysdig](https://github.com/Homebrew/homebrew-core/blob/master/Formula/sysdig.rb)
The test was successful. I write the result below.

```bash
> brew audit sysdig
sysdig:
  * 0.26.5 is a GitHub prerelease
Error: 1 problem in 1 formula detected
```

**Comment by MikeMcQuaid** [comment]
Thanks so much again for your contribution! Without people like you submitting PRs we couldn't run this project. You rock, @hyuraku!

**Comment by iMichka** [comment]
Hi.

I am going to revert this change:
https://github.com/Homebrew/brew/pull/7205

It broke brew audit, for example `brew audit octant --online` fails with `Error: Not Found`.
This probably needs more testing.

**Comment by MikeMcQuaid** [comment]
@hyuraku Sorry we had to revert! Would you be able to look at addressing the mentioned issue and reopening a PR with fixes? Even if not: let me know, thanks!

**Comment by hyuraku** [comment]
@MikeMcQuaid 
I would like to submit another PR about the issue whether GitHub prerelease or not.

**Comment by iMichka** [comment]
Yes, please. Bugs happen, I think if we test it a little bit more before shipping this will be OK.

**Pull request merged**